README.rst

```diff
@@ -334,10 +334,119 @@ This recipe is a little different from other recipes in that it's just a script
+- Setting certain environment variables at container create time
+
+This recipe will attempt to do all of these things in as few steps as possible, except setting environment variables at container create time which you will have to do manually.
+
+---------------------
+Environment variables
+---------------------
+
+Because of how the nvidia runtime operates, it needs certain variables set at container create time. Some ways this can be accomplished include:
+
+- Adding an ``ENV`` to the Dockerfile
+- Adding to the ``environment:`` section in ``docker-compose.yml``
+- Adding the ``-e`` flag to the ``docker run`` call
+
+``NVIDIA_VISIBLE_DEVICES`` must be set, or else the nvidia runtime will not activate, and you will have no gpu support. It's suggested to make set ``NVIDIA_VISIBLE_DEVICES`` to ``all`` and allow for an environment variable on the host (i.e. ``local.env``) to override the value to use specific GPUs as needed.
+
+``NVIDIA_DRIVER_CAPABILITIES`` is usually set, but it is optional because unset and set to null means the ``compute,utility`` capabilities are passed in. This is enough for most CUDA capabilities.
+
+You can optionally add ``NVIDIA_REQUIRE_*`` environment variables (Commonly: ``NVIDIA_REQUIRE_CUDA``) as a set of rules to declare what version of cuda, drivers, architecture, and brand. You only need to set this if you want docker to refuse to run when constraints are not met (e.g. driver doesn't support CUDA 11.6). See the `documentation <https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/user-guide.html#constraints>`_ and an `example <https://gitlab.com/nvidia/container-images/cuda/-/blob/5bc8c5483115b8e9c68d4f1280acb8d56196d681/dist/11.6.2/ubi8/base/Dockerfile#L6>`_ for more information.
+
+----------
+Build Args
+----------
+
+The ``CUDA_VERSION``/``CUDNN_VERSION`` build args must be limited to the versions of CUDA in the `nvidia package repos <https://developer.download.nvidia.com/compute/cuda/repos/>`_. Attempting combinations of OS versions and CUDA versions not in the nvidia `codebase <https://gitlab.com/nvidia/container-images/cuda/-/tree/master/dist>`_ will probably fail because those versions of CUDA most likely do not exist in the nvidia package repos. Currently the end-of-life directory is not supported.
+
+``CUDA_RECIPE_TARGET`` is used to specify how much of CUDA you install (devel vs runtime):
+
+- ``runtime``: Only installs the runtime packages for CUDA
+- ``devel``: Installs both the runtime and development packages for CUDA
+- ``devel-only``: Only installs the devel packages. This is intended to be used on an image that already has the CUDA runtime installed.
+
+.. rubric:: Example
+
+.. code-block:: Dockerfile
+
+   # syntax=docker/dockerfile:1.4
+   FROM vsiri/recipe:cuda as cuda
+
+   FROM redhat/ubi8
+   COPY --from=cuda /usr/local /usr/local
+   # Only needs to be run once for all recipes
+   RUN for patch in /usr/local/share/just/container_build_patch/*; do "${patch}"; done
```

Comment:
it might be worth noting that if using `just` this is done for you

Reply:
You're thinking runtime patches, these are build time. That line is needed